### ISSUE TITLE
Need position_stack_line()

### ISSUE THREAD

**Issue by mikmart** [tidyverse/ggplot2#2883]
`geom_area()` doesn't play well with duplicated `x` values. The docs state:
> `geom_area` is a special case of `geom_ribbon`, where the `ymin` is fixed to 0.

But when there are multiple values of `y` for a given value of `x`, the two are not equivalent. The `geom_ribbon()`  result matches the intuition of "`geom_line()` but with the area under the curve filled in":

``` r
library(tidyverse)
df <- data.frame(x = c(1, 2, 2, 3), y = 1:4)

ggplot(df, aes(x, y)) + geom_ribbon(aes(ymin = 0, ymax = y))
```

![](https://i.imgur.com/DgsipzA.png)

However, it seems this scenario results in `geom_area()` having the equivalent of `ymin = min(y)` and `ymax = sum(y)` in `geom_ribbon()`.

``` r
ggplot(df, aes(x, y)) + geom_area()
```

![](https://i.imgur.com/Qd3ubqt.png)

I'm happy to dig a bit further and see if I can submit a PR for this.

Created on 2018-09-06 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0.9000).

**Comment by mikmart** [comment]
That's true, and perhaps points to a potential issue with `position_stack()` instead:

``` r
library(tidyverse)
df <- data.frame(x = c(1, 2, 2, 3), y = 1:4)
df2 <- bind_rows(a = df, b = df, .id = "g")

ggplot(df2, aes(x, y, fill = g)) + geom_area()
```

![](https://i.imgur.com/nGc52CP.png)

``` r
ggplot(df2, aes(x, y, fill = g)) +
  geom_ribbon(aes(ymin = 0, ymax = y), position = "stack")
```

![](https://i.imgur.com/059wnA4.png)

Created on 2018-09-06 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0.9000).

**Comment by mikmart** [comment]
Then again, `position_stack()` _is_ just doing what it does, stacking `y` values with a common `x`.. i.e. the above would look entirely reasonable with e.g. points. But this doesn't really match the intuition (or mine, at least) for area charts.

**Comment by smouksassi** [comment]
what kind of results you are expecting the above are identical

**Comment by mikmart** [comment]
Yes, that's why this is probably an issue with `position_stack()`. (If it is an issue at all.) I would expect to not have the gaps in between the filled in areas.

**Comment by mikmart** [comment]
Essentially I would want to get this:

``` r
library(tidyverse)

df3 <- data.frame(
  x = c(1, 2, 2, 3, 1, 2, 2, 3),
  y = c(1, 2, 3, 4, 2, 4, 6, 8),
  g = rep(c("b", "a"), each = 4)
)

ggplot(df3, aes(x, y, fill = g)) +
    geom_area(position = "identity")
```

![](https://i.imgur.com/30btz40.png)

This matches my intuition of area stacking, but doesn't really fit all that well with how `position_stack()` works.

Created on 2018-09-06 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0.9000).

**Comment by mikmart** [comment]
The more I think about this the more it feels like perhaps this is not an issue, but rather area stacking in the above manner would be the job for a different position adjustment.

When presenting data like this, only the maximum of the `y` values in a group is really a data point, while the minimum `y` at an `x` value is really just a bounding point for a polygon and doesn't actually represent data.

**Comment by yutannihilation** [comment]
I too feel `geom_area()` seems against the "intuition." I think this feeling comes from the fact that, while we imagine we are stacking areas, we actually are playing with points. For example, if we add a slight jitter to `x` axis, they are not stacked as expected. In the following, you can see `"a"` sinks into `"b"` because it fails to be stacked.

``` r
library(tidyverse)

df <- data.frame(x = rep(1:5, each = 2),
                 y = rep(10 + 1:5, each = 2),
                 g = rep(c("a", "b"), times = 5))
df$x_jitter <- df$x + 10e-4 * runif(10)
df
#>    x  y g x_jitter
#> 1  1 11 a 1.000296
#> 2  1 11 b 1.000814
#> 3  2 12 a 2.000061
#> 4  2 12 b 2.000527
#> 5  3 13 a 3.000078
#> 6  3 13 b 3.000748
#> 7  4 14 a 4.000230
#> 8  4 14 b 4.000891
#> 9  5 15 a 5.000595
#> 10 5 15 b 5.000664

library(egg)
#> Loading required package: gridExtra
#> 
#> Attaching package: 'gridExtra'
#> The following object is masked from 'package:dplyr':
#> 
#>     combine

ggarrange(
  ggplot(df, aes(x, y, fill = g)) + geom_area(),
  ggplot(df, aes(x_jitter, y, fill = g)) + geom_area()
)
```

![](https://i.imgur.com/iBxqLZm.png)

Created on 2018-09-06 by the [reprex package](https://reprex.tidyverse.org) (v0.2.0.9000).

**Comment by mikmart** [comment]
@yutannihilation yeah I think you're right. I read some other issues related to this before posting (#2720, #2802; also just now noticed that this is a duplicate from way back in 2013: #795) but thought this was separate. 

Perhaps it would make sense to have a position adjustment (somewhere) that would handle all of these problems, sort of simulating physical area stacking. That would require automatic interpolation between groups, too, to address the misaligned `x` values.

**Comment by smouksassi** [comment]
yes stacking time series can be done in several ways for example:
there is : https://github.com/AtherEnergy/ggTimeSeries
and this paper.
http://leebyron.com/streamgraph/stackedgraphs_byron_wattenberg.pdf
it may be a special stat on how to handle duplicate x

**Comment by mikmart** [comment]
The Byron & Wattenberg paper is really cool, but I'm not sure it applies to duplicated `y`'s within `x`'s. It's essentially a stacked position adjustment with a varying baseline function rather than a fixed 0 baseline + smart ordering.

**Comment by ptoche** [comment]
@mikmart, is this a new issue or was introduced by an update? I ask out of curiosity because when I filed [2803](https://github.com/tidyverse/ggplot2/issues/2803), I noticed the stacking behaviour had changed over the previous approx 2 years.

**Comment by mikmart** [comment]
I'm not sure, but I would assume it's been like this for quite a while. The way the stacking algorithm works (replacing `y` values that share an `x` value with their cumulative sum) seems pretty fundamental.

**Comment by yutannihilation** [comment]
I've played around this and now feel there's a lot to consider to stack areas properly. I'm afraid this is a bit too complicated to implement in ggplot2...

https://gist.github.com/yutannihilation/2d3851adc874a02f42914f1655329c71

**Comment by yutannihilation** [comment]
Closed by #4889

``` r
library(tidyverse)

df <- data.frame(x = c(1, 2, 2, 3), y = 1:4)
df2 <- bind_rows(a = df, b = df, .id = "g")

ggplot(df2, aes(x, y, fill = g)) + geom_area()
```

![](https://i.imgur.com/trFWQ9f.png)

<sup>Created on 2022-07-07 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
